generators/java/sdk/src/main/java/com/fern/java/client/Cli.java

```diff
@@ -198,19 +198,21 @@ public GeneratedRootClient generateClient(ClientGeneratorContext context, Interm
         generatedTypes.getTypes().values().forEach(this::addGeneratedFile);
         generatedTypes.getInterfaces().values().forEach(this::addGeneratedFile);
 
-        Optional<OAuthScheme> maybeOAuthScheme = context.getIr().getAuth().getSchemes().stream()
-                .map(AuthScheme::getOauth)
-                .flatMap(Optional::stream)
-                .findFirst();
+        Optional<GeneratedJavaFile> generatedOAuthTokenSupplier = Optional.empty();
+        if (context.getGeneratorConfig().getGenerateOauthClients()) {
+            Optional<OAuthScheme> maybeOAuthScheme = context.getIr().getAuth().getSchemes().stream()
```

Comment:
should i throw if someone has oauth configured but `context.getGeneratorConfig().getGenerateOauthClients()` is false?

as written, just doesn't generate the oauth stuff

Reply:
good question i think not generating is okay or falling back to bearer, @amckinney what do you do in the go / ts generators?

Reply:
looks to me like we don't throw in TS:
https://github.com/fern-api/fern/blob/main/generators/typescript/sdk/generator/src/SdkGenerator.ts#L382

couldn't find code falling back to bearer but not 100%, would need @amckinney to confirm

Reply:
Discussed offline - 

We'll go with falling back to bearer in all generators. We'll follow-up with test cases / additional handling in all of the generators that support OAuth.